Add Transparent Pixel Support and Minor Fixes.

Stattek opened the pull request:
Adds ability to render PNGs with transparent pixels (pixels that are at least 95% transparent will be rendered with character at index 0 in charset).
- Add new option to allow ANSI codes to be generated per character rather than when the color has changed, to make it easier to parse by other programs.
- Add new charset that gives similar output to `jp2a` program.
- Change default image output to 128 from 80, to be accurate with documentation.
- Update readme.

## Summary by Sourcery

New Features:
- Support transparent pixels in PNG images, rendering pixels with at least 95% transparency with the minimum index in the charset.

Stattek committed:
Run `cargo fmt`.

Stattek committed:
Allow escaping each colored character individually.

Stattek committed:
Update name of variable.
- Add builder function.

Stattek committed:
Update readme.

Stattek committed:
Update image version.

Stattek committed:
Update cargo lock.

Stattek committed:
Add a new charset that is similar to the output from `jp2a`.

Stattek committed:
Get it closer to jp2a.

Stattek committed:
Merge pull request #1 from Stattek/single-char-escape-sequences

Single char escape sequences

Stattek committed:
Allow displaying transparent pixels.

Stattek committed:
Merge pull request #2 from Stattek/display-transparent-pixels

Allow displaying transparent pixels.

Stattek committed:
Update function and var naming.

Stattek committed:
Update default width to be accurate with comments.

Stattek committed:
Update README.md

Co-authored-by: sourcery-ai[bot] <58596630+sourcery-ai[bot]@users.noreply.github.com>

Stattek committed:
Remove magic number.

orhnk commented on src/image_renderer.rs:
```diff
@@ -24,6 +29,10 @@ impl ImageRenderer<'_> {
     }
 
+    fn get_transparency_percent(&self, pixel: &Rgba<u8>) -> f64 {
+        pixel[3] as f64 / u8::MAX as f64
```
using u8::MAX is smarter than plain 255.

> Just a note to myself

orhnk commented:
Thank you for your PR. I'm waiting for the question

Stattek committed:
Update documentation.

Stattek commented:
> Thank you for your PR. I'm waiting for the question

Is there a question you wanted me to answer? I do not see it, if so.

orhnk commented:
Check out my review please

Stattek committed:
Merge remote-tracking branch 'upstream/master'

Stattek committed:
Remove magic number.

Stattek commented:
> Check out my review please

I have fixed merge conflicts and changed the description.

I do not see a review of what code I should change, maybe the comments weren't uploaded? The only comment you left on my code (as far as I can see) is a note for yourself about `u8::MAX`.

Thank you.
